core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala

```diff
@@ -106,33 +127,15 @@ object HealthMonitor {
   * the /status API call. Instead, the call to /status will return quickly and report there is a problem
   * with the DB (but not other subsystems because the checks are independent).
   *
-  * @param slickDataSource the slick data source for DB operations
-  * @param googleServicesDAO the GCS DAO for Google API calls
-  * @param googlePubSubDAO the GPS DAO for Google PubSub API calls
-  * @param methodRepoDAO the method repo DAO for Agora calls
-  * @param samDAO the IAM DAO for Sam calls
-  * @param executionServiceServers the execution service DAOs for Cromwell calls
-  * @param groupsToCheck Set of Google groups to check for existence
-  * @param topicsToCheck Set of Google PubSub topics to check for existence
-  * @param bucketsToCheck Set of Google bucket IDs to check for existence
+  * @param subsystemChecks list of subsystems and their respective check functions
   * @param futureTimeout amount of time after which subsystem check futures will time out (default 1 minute)
   * @param staleThreshold amount of time after which statuses are considered "stale". If a status is stale
   *                       then it won't be returned to the caller; instead a failing status with an "Unknown"
   *                       message will be returned. This shouldn't normally happen in practice if we have
   *                       reasonable future timeouts; however it is still a defensive check in case something
   *                       unexpected goes wrong. Default 15 minutes.
   */
-class HealthMonitor private (val slickDataSource: SlickDataSource,
-                             val googleServicesDAO: GoogleServicesDAO,
-                             val googlePubSubDAO: GooglePubSubDAO,
-                             val methodRepoDAO: MethodRepoDAO,
-                             val samDAO: SamDAO,
-                             val billingProfileManagerDAO: BillingProfileManagerDAO,
-                             val workspaceManagerDAO: WorkspaceManagerDAO,
-                             val executionServiceServers: Map[ExecutionServiceId, ExecutionServiceDAO],
-                             val groupsToCheck: Seq[String],
-                             val topicsToCheck: Seq[String],
-                             val bucketsToCheck: Seq[String],
+class HealthMonitor private (subsystemChecks: List[(Subsystem, ExecutionContext => Future[SubsystemStatus])],
```

Comment:
this was changed to just take a list of checks to perform, the checks required differ based on whether rawls is running on Google or Azure